### DISCUSSION TITLE
Change Privacy Mode screen

### DISCUSSION THREAD

**Discussion by IndieIndieDev** [rustdesk/rustdesk#2919] [Q&A]
Hello,

I've been looking at changing how the Privacy Mode screen is, but i cannot find where i can edit what is shown as the Privacy Screen.

Any help would be appreciated

**Comment by ccben87** [comment]
I also second this. I would like privacy mode to work the same as in Anydesk where it is a blank, black, screen. I don't want to broadcast to anyone who walks by my computer that I'm remote connected to it.

**Comment by buestad** [comment]
I dived into the code and did som digging and figured out how to solve this myself!
The code for this is inside a dependecy repos written in c++ which produces the WindowInjection.dll inside the RustDesk main folder.

Note: this is for Windows only!

If anyone wants to use my new discrete privacy screen just head over to my fork on GitHub (https://github.com/buestad/RustDeskTempTopMostWindow/releases/tag/v0.2) and download the v0.2 release and replace your WindowInjection.dll with the one inside the zip-file.

If you want your own custom screen you will need to build this .dll yourself since the picture is actually a byte array inside one of the .cpp source files

**Reply by Vasilis553** [reply]
How Can I Actually Do That Change Is There Any Tutorial Or Steps And What Apps You Used To Make it..?

**Reply by buestad** [reply]
If you want to build your own custom screen, use Visual Studio and clone my repos (or the source repos). If you just want to use my version you can just download the zip file in the link and replace the .dll Inside your RustDesk root folder with the one in the zip file

**Reply by Vasilis553** [reply]
I made A blackscreen image to bytes then i changed them in visual studio i builed it and the privacy mode come white and with error message

**Reply by buestad** [reply]
If I remember correctly here is how I did it:
- Use VS 2019 (using 2022 will upgrade some stuff and the built .dll might not work together with other RustDesk binaries?)
- Rebuild the solution for Release x64
- Make a BMP image in 1920x1080 resolution
- Run: `RustDeskTempTopMostWindow\x64\Release>Img2Mem.exe load your_image.bmp`
- Rebuild the solution one more time (now with a modified img.cpp)
- Run: `RustDeskTempTopMostWindow\x64\Release\TestApp.exe`. This should show the new modified screen
- Copy the `RustDeskTempTopMostWindow\x64\Release\WindowInjection.dll` to `C:\Program Files\RustDesk` overwriting the old one

**Reply by Vasilis553** [reply]
Thanks A lot for helping me make this change and because you created the article for help people!!

**Reply by retardioremilio** [reply]
Is this possible when using the pro version custom client?
Im not sure where it's pulling the build from.

**Reply by osafbty** [reply]
@Vasilis553 did you manage to get it working?

**Comment by Ivymike1952** [comment]
can you explain better step by step?

**Reply by buestad** [reply]
What do you need more detail on? If you have never built a Visual Studio project before check out some YouTube videos, or ask ChatGPT 😊👍

**Reply by Ivymike1952** [reply]
Done it! But i have a problem, on one notebook when enable privacy mode, i can see my blackscreen but in both pc, not only remote pc so i can't doing anywhere.
In older notebook, i can use only Mode Privacy 2 where it showing default blackscreen, and Mode Privacy 1 is not available.

**Reply by 333WXY** [reply]
> can you explain better step by step?

I encountered the same problem. Both computers went black

**Comment by southerndoc** [comment]
Would be nice if there was an option to make the screen blank instead of privacy notice.  Hopefully a future release will offer it.

Even better would be a customizable option (e.g., choose black/blank or be able to customize text -- maybe a company wants to put a phone number for someone to call if they need their computer).

**Comment by southerndoc** [comment]
Just realized this is the same request as here:  https://github.com/rustdesk/rustdesk/discussions/10488

Maybe the mods can combine the threads.

**Comment by southerndoc** [comment]
@rustdesk Please make this a priority to develop a blank privacy screen AND a privacy screen for macOS.

**Comment by niwiunig** [comment]
Thank you @buestad. I have just successfully built a privacy screen with VS2022 (Because i don't have 2019 installed and was not able to download it) but what is happening is that the privacy screen even shows in the connection. Did anyone else have this problem and was able to fix this?

**Comment by OmarRC64** [comment]
Still no update for this? Would be a great feature to have.

**Comment by southerndoc** [comment]
Glad to see the privacy screen on macOS servers now...  There is a bug where it doesn't lock after session end when the privacy screen is enabled.
